### PR TITLE
feat: move agents to new tile

### PR DESCRIPTION
**Pull request by jor2**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: ibm_catalog.json [hunk 1]
```diff
@@ -0,0 +1,209 @@
+{
+    "products": [
+      {
+        "name": "deploy-arch-ibm-scc-workload-protection-agent",
```
Comment (ocofaigh) [inline ibm_catalog.json:4]: maybe go with `deploy-arch-ibm-scc-wp-agent`

File: ibm_catalog.json [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        "label": "Cloud automation for Security and Compliance Center Agents",
```
Comment (ocofaigh) [inline ibm_catalog.json:5]: It is a bit long, but I think the correct working would be:
"Cloud automation for Security and Compliance Center Workload Protection agent"

File: ibm_catalog.json [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+          "SCC",
+          "security",
+          "compliance",
+          "workload-protection",
+          "IaC",
+          "infrastructure as code",
+          "terraform",
+          "solution"
+        ],
+        "short_description": "Creates and configures IBM Security and Compliance Center resources",
```
Comment (ocofaigh) [inline ibm_catalog.json:24]: Configures Security and Compliance Center Workload Protection agents

File: ibm_catalog.json [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        "long_description": "This architecture supports creating and configuring IBM Security and Compliance Center resources.",
```
Comment (ocofaigh) [inline ibm_catalog.json:25]: This architecture supports creating and configuring IBM Security and Compliance Center Workload Protection agent to an IBM Cloud Kubernentes or Openshift cluster

File: ibm_catalog.json [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+        "offering_docs_url": "https://github.com/terraform-ibm-modules/terraform-ibm-scc-da/blob/main/solutions/instances/README.md",
```
Comment (ocofaigh) [inline ibm_catalog.json:26]: needs to change to new location

File: ibm_catalog.json [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+        "offering_icon_url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-scc-da/main/images/scc-icon.svg",
```
Comment (ocofaigh) [inline ibm_catalog.json:27]: suggest to move the icon too

File: ibm_catalog.json [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+          {
+            "title": "Security and Compliance Center Workload Protection Agents",
+            "description": "Creates and configures the Workload Protections components in an existing cluster to an existing IBM Cloud Security and Compliance Center Workload Protection instance."
+          }
+        ],
+        "support_details": "This product is in the community registry, as such support is handled through the originated repo. If you experience issues please open an issue in that repository [https://github.com/terraform-ibm-modules/terraform-ibm-scc-da/issues](https://github.com/terraform-ibm-modules/terraform-ibm-scc-da/issues). Please note this product is not supported via the IBM Cloud Support Center.",
+        "flavors": [
+          {
+            "label": "Agents",
+            "name": "agents",
```
Comment (ocofaigh) [inline ibm_catalog.json:39]: We no longer need to call it agents. Suggest to go with:

"label": "Standard",
"name": "standard",

File: ibm_catalog.json [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+            "install_type": "extension",
```
Comment (ocofaigh) [inline ibm_catalog.json:40]: change this to "fullstack" and add 
`"dependency_version_2": true,`

File: ibm_catalog.json [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+            "working_directory": "solutions/agents",
```
Comment (ocofaigh) [inline ibm_catalog.json:41]: I guess move the code to a `standard` directory to match the new proposed variation name

File: ibm_catalog.json [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+                ]
+              }
+            ],
+            "iam_permissions": [
+              {
+                "service_name": "compliance",
+                "role_crns": [
+                  "crn:v1:bluemix:public:iam::::serviceRole:Manager",
+                  "crn:v1:bluemix:public:iam::::role:Editor"
+                ]
```
Comment (ocofaigh) [inline ibm_catalog.json:176]: these permissions are not needed for agents

File: ibm_catalog.json [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+                {
+                  "title": "Security and Compliance Center Workload Protection agent",
+                  "description": "Creates and configures a Security and Compliance Center Workload Protection agent."
+                }
+              ],
+              "diagrams": [
+                {
+                  "diagram": {
+                    "caption": "Security and Compliance Center Workload Protection agent",
+                    "url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-scc-da/main/reference-architecture/scc-wp-agent.svg",
```
Comment (ocofaigh) [inline ibm_catalog.json:198]: update diagram link to this repo

**Review by ocofaigh — CHANGES_REQUESTED** [review]
See comments. Can you also:
- add a `.catalog-onboard-pipeline.yaml` (Ill share the offering Id after next commit to this PR)
- update .releaserc to trigger the catalog pipeline

**Comment by jor2** [comment]
> See comments. Can you also:
> 
> * add a `.catalog-onboard-pipeline.yaml` (Ill share the offering Id after next commit to this PR)
> * update .releaserc to trigger the catalog pipeline

updated and added missing scripts

File: .catalog-onboard-pipeline.yaml [hunk 12]
```diff
@@ -0,0 +1,13 @@
+---
+apiVersion: v1
+offerings:
+  - name: deploy-arch-ibm-scc-wp-agent
+    kind: solution
+    catalog_id: 7df1e4ca-d54c-4fd0-82ce-3d13247308cd
+    offering_id: 9423f9bc-1290-4c71-a9ac-01898bfa7ccc
```
Comment (ocofaigh) [inline .catalog-onboard-pipeline.yaml:7]: @jor2 change this to `f76f2ed3-eb98-4e3d-b147-360485fd8a36`

**Review by ocofaigh — DISMISSED** [review]

File: tests/pr_test.go [hunk 13]
```diff
@@ -2,16 +2,37 @@
...
+	"golang.org/x/exp/rand"
 )
 
+const resourceGroup = "geretain-test-resources"
+
 // const resourceGroup = "geretain-test-resources"
 const basicExampleDir = "examples/basic"
 const secureExampleDir = "examples/secure"
+const agentFlavorDir = "solutions/agents"
+const agentsKubeconfigDir = "solutions/agents/kubeconfig"
```
Comment (ocofaigh) [inline tests/pr_test.go:27]: this need to be updated to /solutions/standard. I think thats why the test failed

Comment (jor2) [reply tests/pr_test.go:27]: oh :(

**Comment by ocofaigh** [comment]
@jor2 you need to resolve conflicts before pipeline can be triggered

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**